### ISSUE TITLE
BUG: implementation ToPaginatedListAsync with PaginationSpecification only pick up the last condition in condition list

### ISSUE THREAD

**Issue by NotAsea** [TanvirArjel/EFCore.GenericRepository#45]
In this method [QueryableExtensions.ToPaginatedListAsync](https://github.com/TanvirArjel/EFCore.GenericRepository/blob/01a306dc7ed892806d669cdbc183138ad721c068/src/TanvirArjel.EFCore.QueryRepository/QueryableExtensions.cs#L74`)
this block code is problem
```cs
IQueryable<T> countSource = source;

// modify the IQueryable using the specification's expression criteria
 if (specification.Conditions != null && specification.Conditions.Count != 0)
{
     foreach (Expression<Func<T, bool>> condition in specification.Conditions)
     {
           countSource = source.Where(condition);
     }
}
```
as you can see the current loop will always assign the current condition in condition list to `countSource `, which override every item before it. I guess people usually use one condition then everything work fine, but when testing with multiple condition i observe something off and browsing from source, i found this. 
I think this is small problem due to you mistyping :)), anyway it should be 
```patch
-  countSource = source.Where(condition);
+ countSource = countSource.Where(condition);
```

**Comment by NotAsea** [comment]
also someone already got this bug #29

**Comment by TanvirArjel** [comment]
@NotAsea Thank you for noticing it. I shall take of it. Or You can request an MR.

**Comment by TanvirArjel** [comment]
The issue has been fixed in versions `6.04` and `1.5.3`!
